addons/mrp/models/mrp_bom.py

```diff
@@ -104,6 +104,11 @@ def _compute_possible_product_template_attribute_value_ids(self):
         for bom in self:
             bom.possible_product_template_attribute_value_ids = bom.product_tmpl_id.valid_product_template_attribute_line_ids._without_no_variant_attributes().product_template_value_ids._only_active()
 
+    @api.depends('company_id')
+    def _compute_currency_id(self):
+        for bom in self:
+            bom.currency_id = bom.company_id.sudo().currency_id.id or self.env.company.currency_id.id
```

Comment:
You really need the sudo? You are not suppose to access BoM from company you don't belong to. Neither be able to select it